test/int/cluster_test.go

```diff
@@ -68,72 +69,86 @@ var _ = Describe("AtlasCluster", func() {
-			Expect(atlasCluster.Name).To(Equal(expectedCluster.Spec.Name))
-			Expect(atlasCluster.ProviderSettings.InstanceSizeName).To(Equal(expectedCluster.Spec.ProviderSettings.InstanceSizeName))
-			Expect(atlasCluster.ProviderSettings.ProviderName).To(Equal(expectedCluster.Spec.ProviderSettings.ProviderName))
-			Expect(atlasCluster.ProviderSettings.RegionName).To(Equal(expectedCluster.Spec.ProviderSettings.RegionName))
+				// Unfortunately we cannot do global checks on cluster/providerSettings fields as Atlas adds default values
+				Expect(atlasCluster.Name).To(Equal(expectedCluster.Spec.Name))
+				Expect(atlasCluster.ProviderSettings.InstanceSizeName).To(Equal(expectedCluster.Spec.ProviderSettings.InstanceSizeName))
+				Expect(atlasCluster.ProviderSettings.ProviderName).To(Equal(expectedCluster.Spec.ProviderSettings.ProviderName))
+				Expect(atlasCluster.ProviderSettings.RegionName).To(Equal(expectedCluster.Spec.ProviderSettings.RegionName))
+			})
 
 			// TODO check connectivity to cluster
 
-			By("Updating the Cluster")
-			createdCluster.Spec.Labels = []mdbv1.LabelSpec{{Key: "int-test", Value: "true"}}
-			Expect(k8sClient.Update(context.Background(), createdCluster)).To(Succeed())
-
-			Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterUpdatingFunc()),
-				1200, interval).Should(BeTrue())
-
-			Expect(createdCluster.Status.ConnectionStrings).NotTo(BeNil())
-			Expect(createdCluster.Status.ConnectionStrings.Standard).NotTo(BeNil())
-			Expect(createdCluster.Status.ConnectionStrings.StandardSrv).NotTo(BeNil())
-			Expect(createdCluster.Status.MongoDBVersion).NotTo(BeNil())
-			Expect(createdCluster.Status.MongoURIUpdated).NotTo(BeNil())
-			Expect(createdCluster.Status.StateName).To(Equal("IDLE"))
-			Expect(createdCluster.Status.Conditions).To(ConsistOf(expectedConditionsMatchers))
-			Expect(createdCluster.Status.ObservedGeneration).To(Equal(createdCluster.Generation))
-
-			// Atlas
-			atlasCluster, _, err = atlasClient.Clusters.Get(context.Background(), createdProject.Status.ID, createdCluster.Spec.Name)
-			Expect(err).ToNot(HaveOccurred())
-
-			createdAtlasCluster, err := createdCluster.Spec.Cluster()
-			Expect(err).ToNot(HaveOccurred())
-
-			Expect(atlasCluster.Name).To(Equal(createdAtlasCluster.Name))
-			Expect(atlasCluster.Labels).To(Equal(createdAtlasCluster.Labels))
-			Expect(atlasCluster.ProviderSettings.InstanceSizeName).To(Equal(createdAtlasCluster.ProviderSettings.InstanceSizeName))
-			Expect(atlasCluster.ProviderSettings.ProviderName).To(Equal(createdAtlasCluster.ProviderSettings.ProviderName))
-			Expect(atlasCluster.ProviderSettings.RegionName).To(Equal(createdAtlasCluster.ProviderSettings.RegionName))
+			By("Updating the Cluster labels", func() {
+				createdCluster.Spec.Labels = []mdbv1.LabelSpec{{Key: "int-test", Value: "true"}}
+				performUpdate()
```

Comment:
I think to make the test more reliable we need to add checks for AtlasCluster state in Atlas after each update. Thus we'll make sure we have changed the new field and haven't changed the others. We can move the common checks (providerSettings etc) to another function like you did for "doCommonChecks" and call that.

Maybe for making this more concise even create a function "checkInAtlas" and pass the callback to assert the object in Atlas leaving out boilerplate code of "GET" requests.

WDYT?